utils/sharedcache/factory.go

```diff
@@ -0,0 +1,27 @@
+package sharedcache
+
+import (
+	"fmt"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+const (
```

Comment:
I think it is better to do enums like:

```go
type Cache int

const (
	CacheMutable Cache = iota
	CacheImmutable
)
```
as it is more explicit and I think it means that you can't accidentally use a value that is out of range. You don't want people doing `NewCache(1, ....)` because if someone added `const x = 1` somewhere then `iota` will change the value of `CacheMutable` to avoid conflicts. Hence if we create the `type Cache` we are abstracting the enum away from being an int so people will have to go out of their way to use ints and they won't accidentally do something they shouldn't